### PR TITLE
Release of 7.3.0

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Preparation for release 7.3.0
|New feature| No 
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by PetrHeinz** [comment]
Removing or changing msgids of translation messages is considered a BC break and, as such, may be done only in a `MAJOR` release. See [our BC Promise for Translation Messages](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md#translation-messages)

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,81 @@ The changelog is generated during the release process using [ChangelogLinker](ht
 
 <!-- changelog-linker -->
 
+## [v7.3.0](https://github.com/shopsys/shopsys/compare/v7.2.2...v7.3.0) - 2019-07-04
```
Comment (vitek-rostislav) [inline CHANGELOG.md]: Uf, I had a quite hard time to review the changelog :slightly_frowning_face: 
Thanks for providing me the original dump and information what you have added manually :+1:  - I have some thoughts here :smile: 
- I do agree with not including changes that just fixed something on the master (BC compatibility improvements like #1131, #1143, etc.) so they are not relevant in the context of viewing the changes since the previous release
- I rather agree with not including typo-like fixes (#1160, #1080, ...), however, I do not like the fact that we need to think about it...
- I disagree with not including #1158, #1172, #1162, and I am not sure about the other not included changes - #1130, #1179, #1128...

So, to sum it up, I think you should include #1158, #1172, #1162 into the changelog and I think we really need to do something with the automatic dumping to make it work with multiple branches better...there is too much manual work in this step at the moment :disappointed:

File: packages/framework/build.xml [hunk 2]
```diff
@@ -78,7 +78,7 @@
         <echo level="warning" message="This phing target is deprecated since Shopsys Framework 7.3.0, use 'build-demo-ci' instead."/>
     </target>
 
-    <target name="build-demo-dev" depends="build-version-generate,wipe-excluding-logs,composer-dev,npm,timezones-check,dirs-create,test-dirs-create,assets,db-demo,product-search-recreate-structure,product-search-export-products,grunt,error-pages-generate,tests-acceptance-build,clean-redis-old,checks-diff" description="Builds application for development with clean demo DB and runs checks on changed files."/>
+    <target name="build-demo-dev" depends="build-version-generate,wipe-excluding-logs,composer-dev,npm,timezones-check,dirs-create,test-dirs-create,assets,db-demo,product-search-recreate-structure,product-search-export-products,grunt,error-pages-generate,tests-acceptance-build,clean-redis-old,checks" description="Builds application for development with clean demo DB and runs checks on changed files."/>
```
Comment (vitek-rostislav) [inline packages/framework/build.xml]: Hm - the build will take more time now, I do like that :disappointed: 

I thought the Windows/Mac problem is covered by the info messages added in https://github.com/shopsys/shopsys/commit/fc50b78c9832a61faad8679ed5b3d1215e2ca51d

However, if this is the way we want to go (if I am outvoted by the rest of the team), you need to fix the description so it is not talking about the changed files anymore :wink:

File: docs/upgrade/phpstan-level-4.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
 # Upgrade Instructions for Upgrading PHPStan to Level 4
 
 This article describes the upgrade instructions for [#1040 Upgrading PHPStan to level 4](https://github.com/shopsys/shopsys/pull/1040).
-Upgrade instructions are in a separate article because there is a lot of instructions and we don't want to jam the [UPGRADE-unreleased.md](/docs/upgrade/UPGRADE-unreleased.md). <!--- TODO change to released version -->
-Instructions are meant to be followed when you upgrade from `v7.2.0` to `unreleased`. <!--- TODO change to released version -->
+Upgrade instructions are in a separate article because there is a lot of instructions and we don't want to jam the [UPGRADE-v7.3.0](/docs/upgrade/UPGRADE-v7.2.2.md).
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/upgrade/phpstan-level-4.md]: Ou... I probably copied wrong version string. Fixed

File: packages/framework/src/Resources/translations/messages.cs.po [hunk 4]
```diff
@@ -388,9 +388,6 @@ msgstr "Kód"
 msgid "Code of service Heureka - Verified by Customer"
 msgstr "Kód služby Ověřeno zákazníky"
 
-msgid "Coefficient"
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply packages/framework/src/Resources/translations/messages.cs.po]: You're both right, I didn't realize this. Commit removed.

[the remark this answers is not in the document]
Comment (grossmannmartin) [reply packages/framework/src/Resources/translations/messages.cs.po]: However I'm thinking about some bc safe version of `dump-translations`

File: packages/framework/build.xml (hunk 2 above)
Comment (grossmannmartin) [reply packages/framework/build.xml]: Well, imagine a situation you want to try Framework on Mac, you don't know anything about it and you run the installation script and the outcome of this is "error do something". It doesn't feel right :(

I didn't know we (the team) already had this discussion. I'm thinking about some conditions or something like that...

But currently the installation simply does not work on Mac.

Comment (henzigo) [reply packages/framework/build.xml]: Maybe, we should use `build-demo-dev-quick` in installation guides, because probably people do not want run tests when they first install that.

File: CHANGELOG.md (hunk 1 above)
Comment (grossmannmartin) [reply CHANGELOG.md]: About #1158 – It's not included intentionally because the problem was solved more systematically in #1161 and the solution from 1158 is not in the codebase anymore.
#1172 – okay, added
#1162 – hmm... okay, this information may be beneficial to the users. Added.

And I don't think the fixes and improvements in documentation should be a part of changelog (especially when it's only one sentence) – #1130, #1179, #1128

**Comment by grossmannmartin** [comment]
> Removing or changing msgids of translation messages is considered a BC break and, as such, may be done only in a `MAJOR` release. See [our BC Promise for Translation Messages](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md#translation-messages)

You're right. I removed the commit.
I'm thinking about some bc safe version of `dump-translations` that doesn't remove translations and is executed when you releasing minor version.

File: packages/framework/build.xml (hunk 2 above)
Comment (grossmannmartin) [reply packages/framework/build.xml]: Running `build-demo-dev-quick` and `error-pages-generate` in the installation script and mention this in the installation guide seems like the okayest solution right now.
What do you say, @vitek-rostislav?

Comment (boris-brtan) [reply packages/framework/build.xml]: oh daaamn we forgot about this https://github.com/shopsys/shopsys/commit/fc50b78c9832a61faad8679ed5b3d1215e2ca51d#diff-3c2478376ba4cacfb00c91ceb062a5a2
this should be also mentined in upgrade

File: packages/framework/src/Resources/translations/messages.cs.po (hunk 4 above)
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply packages/framework/src/Resources/translations/messages.cs.po]: Yes, this would be nice, I think you can just simply use `--keep=true` parameter to achieve this

File: CHANGELOG.md (hunk 1 above)
Comment (vitek-rostislav) [reply CHANGELOG.md]: Thanks for the answer and fixes :ok_hand:

File: packages/framework/build.xml (hunk 2 above)
Comment (grossmannmartin) [reply packages/framework/build.xml]: As the build.xml file is now in framework, I don't think the upgrade notes are necessary.

Comment (vitek-rostislav) [reply packages/framework/build.xml]: I like it :slightly_smiling_face:

**Comment by vitek-rostislav** [comment]
project-base docker installation tested on Ubuntu, everything works :ok_hand: :heavy_check_mark:

File: docs/installation/installation-using-docker-application-setup.md [hunk 5]
```diff
@@ -61,4 +61,4 @@ Elasticsearch API is available on the address [http://127.0.0.1:9200](http://127
 You can use e.g. [Postman](https://www.getpostman.com/apps) or [Kibana](https://www.elastic.co/downloads/kibana) for Elasticseacrh management.
 
 If you need to inspect your application logs, use `docker-compose logs` command.
-For more information about logging see [the separate article](/docs/introduction/logging.md).
\ No newline at end of file
+For more information about logging see [the separate article](/docs/introduction/logging.md).
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/installation/installation-using-docker-application-setup.md]: Wanted :) Files should end with a new line

File: docs/installation/installation-using-docker-application-setup.md [hunk 6]
```diff
@@ -41,7 +41,7 @@ More information about what Phing targets are and how they work can be found in
 
 ### 1.4. Build the application
 ```
-php phing build-demo-dev
+php phing build-demo-dev-quick error-pages-generate
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/installation/installation-using-docker-application-setup.md]: To me, this instruction doesn't seem connected – `timezones-check` checks timezones in PHP and PostgreSQL only, and I don't see anywhere passing the timezone to Docker. And on Mac, there is no such thing. 
Maybe @boris-brtan can confirm or deny this.

So right now, I would prefer to not change Phing targets and maybe create isolated PR with the changes.

[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/installation/installation-using-docker-application-setup.md]: I changed the commit message, though.

File: project-base/scripts/install.sh [hunk 7]
```diff
@@ -75,7 +75,7 @@ echo "Installing application inside a php-fpm container"
 
 docker-compose exec php-fpm composer install
 
-docker-compose exec php-fpm ./phing db-create test-db-create build-demo-dev
+docker-compose exec php-fpm ./phing db-create test-db-create build-demo-dev-quick error-pages-generate
```
Comment (boris-brtan) [inline project-base/scripts/install.sh]: hello guys, here is change in project-base without added upgrade notes ...

Comment (grossmannmartin) [reply project-base/scripts/install.sh]: Wow! Thanks for pointing it out. I didn't realize this. 
I added the upgrade notes.
